optimism/src/mips/witness.rs

```diff
@@ -100,6 +100,100 @@ fn memory_size(total: usize) -> String {
 impl<Fp: Field> InterpreterEnv for Env<Fp> {
     type Variable = u32;
 
+    // Debug
+    fn debug_signed_16bits_variable(v: &Self::Variable) -> String {
+        // FIXME: It does suppose u32 is between 0 and 2 ^ 16
+        let v = *v;
+        // handle 0b1000000000000000
+        if v == 1 << 15 {
+            return String::from("0");
+        }
+        let bit_sign = v >> 15;
+        if bit_sign == 1 {
+            let v = !(v - 1) & 0x0000ffff;
```

Comment:
so it is not one leading bit to indicate sign, and the rest is the binary "absolute value"? Instead, you need to subtract 1 to the value if it is negative, then negate it all, and then get the 16 rightmost bits and that's the absolute value?